sunpy/net/fido_factory.py

```diff
@@ -317,7 +265,17 @@ def search(self, *query):
         parts individually.
         """  # noqa
         query = attr.and_(*query)
-        return UnifiedResponse(query_walker.create(query, self))
+        results = query_walker.create(query, self)
+
+        # If we have searched the VSO but no results were returned, but another
+        # client generated results, we drop the empty VSO results for tidiness.
```

Comment:
This behaviour should be documented somewhere in the user facing docs

I also wondering if it should be generalised - if at least one client returns results, should all other zero-result clients be dropped?

Reply:
VSO is special cased in a way, it's `_can_handle_query` is exceptionally broad (mostly for historical reasons). This therefore is a way of bringing VSO into line with the other clients which afaik don't have their `_can_handle_query` return True for a query they will not return results for.

Reply:
I have updated the comment to elaborate on why.